src/shell/filesystem_shell.rs

```diff
@@ -96,28 +95,37 @@ impl Shell for FilesystemShell {
         }: LsArgs,
         context: &RunnablePerItemContext,
     ) -> Result<OutputStream, ShellError> {
+        let ctrl_c = context.ctrl_c.clone();
+        let name_tag = context.name.clone();
+
         let (path, p_tag) = match path {
             Some(p) => {
                 let p_tag = p.tag;
                 let mut p = p.item;
                 if p.is_dir() {
+                    if std::fs::read_dir(&p).map_or(false, |mut s| s.next().is_none()) {
```

Comment:
I'm not proud of solving it this way but if it's stupid and it works.....

Reply:
It's actually not too bad to read, but if you extract it into a function with a good name, it'll make this code even clearer 💪 

I have to head out right now, but I'm hoping sometime between now and tomorrow I can think about what you've done a little more deeply to ensure all of our bases are covered (I'll check out your code and run it). What's important to me is that we don't cause problems for the simple cases. As long as we can `ls` and not have it fail for something that exists, or matches, we should be good 👍